fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBStoreTimerTest.java

```diff
@@ -0,0 +1,191 @@
+        diffTimer = StoreTimer.getDifference(latestTimer, savedTimer);
+        diffKVs = diffTimer.getKeysAndValues();
+        Assertions.assertTrue(diffKVs.containsKey("load_scan_entry_count"));
+        Assertions.assertEquals(diffKVs.get("load_scan_entry_count").intValue(), numAdvances);
+        Assertions.assertTrue(diffKVs.containsKey("load_key_value_count"));
+        Assertions.assertEquals(diffKVs.get("load_key_value_count").intValue(), numAdvances);
+    }
+
+    @Test
+    public void timeoutCounterDifferenceTest() {
+        RecordCursor<KeyValue> kvc = KeyValueCursor.Builder.withSubspace(subspace).setContext(context).setScanProperties(ScanProperties.FORWARD_SCAN).setRange(TupleRange.ALL).build();
+
+        FDBStoreTimer latestTimer = context.getTimer();
+        StoreTimerSnapshot savedTimer;
+        CompletableFuture<RecordCursorResult<KeyValue>> fkvr;
+        RecordCursorResult<KeyValue> kvr;
+        StoreTimer diffTimer;
+        Map<String, Number> diffKVs;
+
+        // generate a first timeout event advancing the cursor
+        //context.getDatabase().setAsyncToSyncTimeout(1, TimeUnit.MILLISECONDS);
+
+        //record timeout
+        latestTimer.recordTimeout(FDBStoreTimer.Waits.WAIT_ADVANCE_CURSOR, System.nanoTime());
+
+        // the latest timer should have recorded the one timeout event
+        diffKVs = latestTimer.getKeysAndValues();
+        Assertions.assertEquals(diffKVs.get("wait_advance_cursor_timeout_count").intValue(), 1);
+        Assertions.assertTrue(diffKVs.get("wait_advance_cursor_timeout_micros").intValue() > 0);
+
+        //advance the cursor without timing out
+        latestTimer.record(FDBStoreTimer.Waits.WAIT_ADVANCE_CURSOR, System.nanoTime());
+        latestTimer.record(FDBStoreTimer.Waits.WAIT_ADVANCE_CURSOR, System.nanoTime());
+
+        // record the state after the first timeout event and generate some more timeout events
+        savedTimer = StoreTimerSnapshot.from(latestTimer);
+        final int numTimeouts = 3;
+        for (int i = 0; i < numTimeouts; i++) {
+            latestTimer.recordTimeout(FDBStoreTimer.Waits.WAIT_ADVANCE_CURSOR, System.nanoTime());
+        }
+
+        // should have the additional timeout events in latestTimer
+        diffKVs = latestTimer.getKeysAndValues();
+        Assertions.assertEquals(diffKVs.get("wait_advance_cursor_timeout_count").intValue(), numTimeouts + 1);
+        Assertions.assertTrue(diffKVs.get("wait_advance_cursor_timeout_micros").intValue() > 0);
+
+        // the savedTimer should only have recorded the first timeout event and hence the difference is the numTimeout events that occurred after that
+        diffTimer = StoreTimer.getDifference(latestTimer, savedTimer);
+        diffKVs = diffTimer.getKeysAndValues();
+        Assertions.assertEquals(diffKVs.get("wait_advance_cursor_timeout_count").intValue(), numTimeouts);
+        Assertions.assertTrue(diffKVs.get("wait_advance_cursor_timeout_micros").intValue() > 0);
```

Comment:
This line actually failed for me locally. I suspect it could be because it's possible that zero nanos have passed between the `recordTimeout` method being called and then the timeout being added (which subtracts the current time (i.e., `System.nanoTime()`) from the passed start time. Perhaps if the `recordTimeout` call set it's time to `System.nanoTime() - 5000` or something so that it was always at least 5 micros old at the time or something.